File: Idem/Idem.java
import java.util.Scanner;
import java.math.BigInteger;

public class Idem {

    public static BigInteger fac(int f) {
	BigInteger res = BigInteger.ONE;
	while(f > 1) {
	    res = res.multiply(new BigInteger(Integer.toString(f)));
	    --f;
	}
	return res.subtract(BigInteger.ONE);
    }

    public static void main(String[] args) {
	Scanner s = new Scanner(System.in);
	int t = s.nextInt();
	for(int i = 0; i < t; i++) {
	    System.out.println(fac(s.nextInt()).mod(new BigInteger("1000000007")));
	}
    }

}

File: Idem/Idem.cc
#include<iostream>

using namespace std;

int fac(long long f)
{
    long long res = 1;
    while(f > 1) {
	res = (res * f) % 1000000007;
	--f;
    }
}
int main()
{
    int t;
    cin >> t;
    for(int i = 0; i < t; i++) {
	long long n;
	cin >> n;
	cout << (fac(n) + 1000000006) % 1000000007 << endl;
    }
}
